rest/config_manager.go

```diff
@@ -263,21 +276,29 @@ outer:
 		// Persist registry
 		writeErr := b.setGatewayRegistry(ctx, bucketName, registry)
 		if writeErr == nil {
+			base.DebugfCtx(ctx, base.KeyConfig, "DeleteConfig persisted updated registry successfully")
+			registryUpdated = true
 			break
 		}
+
 		// retry on cas mismatch, otherwise return error
 		if !base.IsCasMismatch(writeErr) {
 			base.InfofCtx(ctx, base.KeyConfig, "DeleteConfig failed to write updated registry: %v", writeErr)
 			return writeErr
 		}
 
+		base.InfofCtx(ctx, base.KeyConfig, "DeleteConfig encountered cas mismatch error while persisting updated registry: %v", writeErr)
+
 		// Check for context cancel before retrying
 		select {
 		case <-ctx.Done():
-			break outer
+			break
```

Comment:
Should this return here? I'm not sure if this should be an error or not.

Reply:
I think you're right - updated that to match insert and update.